target_chains/ethereum/contracts/contracts/pulse/PulseState.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+contract PulseState {
+    uint8 public constant NUM_REQUESTS = 32;
+    bytes1 public constant NUM_REQUESTS_MASK = 0x1f;
+
+    struct Request {
+        address provider;
+        uint64 sequenceNumber;
+        uint256 publishTime;
+        bytes32[] priceIds;
```

Comment:
I think this is not stored directly in the request and therefore defeats the optimisation we had to remove SSTORE calls. Maybe a fixed length array or a hash of priceIds? There is of course trade offs

Reply:
good point, I think `priceIdsHash` makes more sense as it has no size limit compared to using fixed length array